### PR TITLE
Шагов М.А. 381806-1. ЛР3

### PR DESCRIPTION
**Pull request by MaksimShagov**
### Pull Request Readiness Checklist

See details at [code review]( https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/main/lab-guide/lab-1-setup.md#%D1%80%D0%B5%D1%86%D0%B5%D0%BD%D0%B7%D0%B8%D1%8F-%D0%BA%D0%BE%D0%B4%D0%B0)

- [x] Code is well documented and more information can be seen on the [website](https://devtools-course-practice-2021.readthedocs.io/en/latest/modules.html) (recommended)
- [x] Make sure that the testing has passed successfully on [GitHub Actions](https://github.com/UNN-ITMM-Software/devtools-course-practice/actions)
- [x] Get two approve from your fellow students
- [ ] Take a review from one of the administrators

### CODE REVIEW

File: modules/vector/include/vector_calculator.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    bool validateNumberOfArguments(int argc, const char** argv);
+    std::string message_;
+    typedef struct {
+        Vector v_1;
+        Vector v_2;
+        char operation;
+    } Arguments;
+};
+
+//! @}
```
Comment (vla5924) [inline modules/vector/include/vector_calculator.h:26]:
```suggestion
```

File: modules/vector/src/vector_calculator.cpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    message_ =
+        std::string(message) +
+          "This is a vector calculator application.\n\n" +
+          "Please provide arguments in the following format:\n\n"+
+
+          "  $ " + appname + " <v1_x> <v1_y> <v1_z> " +
+          "<v2_x> <v2_y> <v2_z> <operation>\n\n" +
+
+          "Where all arguments are double-precision numbers, " +
+          "and <operation> is one of 'norm', 'dot', '*', 'normalize'.\n";
```
Comment (vla5924) [inline modules/vector/src/vector_calculator.cpp:25]:
```suggestion
        std::string(message) +
          "This is a vector calculator application.\n\n" \
          "Please provide arguments in the following format:\n\n" \
          "  $ " + appname + " <v1_x> <v1_y> <v1_z> " \
          "<v2_x> <v2_y> <v2_z> <operation>\n\n" \
          "Where all arguments are double-precision numbers, " \
          "and <operation> is one of 'norm', 'dot', '*', 'normalize'.\n";
```

**Review by vla5924 — CHANGES_REQUESTED** [review]

**Comment by MaksimShagov** [comment]
@taktaev-artyom

File: modules/vector/include/vector_calculator.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+ private:
+    void help(const char* appname, const char* message = "");
+    bool validateNumberOfArguments(int argc, const char** argv);
+    std::string message_;
+    typedef struct {
+        Vector v_1;
+        Vector v_2;
+        char operation;
+    } Arguments;
```
Comment (vla5924) [inline modules/vector/include/vector_calculator.h:23]:
```suggestion
    struct Arguments {
        Vector v_1;
        Vector v_2;
        char operation;
    };
```

**Review by vla5924 — CHANGES_REQUESTED** [review]

**Review by vla5924 — APPROVED** [review]

File: modules/vector/src/vector_calculator.cpp [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+#include <string>
+#include <sstream>
+
+VectorCalculator::VectorCalculator() : message_("") {}
+
+void VectorCalculator::help(const char* appname, const char* message) {
+    message_ =
+        std::string(message) +
+           "This is a vector calculator application.\n\n" \
+          "Please provide arguments in the following format:\n\n" \
```
Comment (taktaev-artyom) [inline modules/vector/src/vector_calculator.cpp:19]: Достаточно будет одного \n

**Review by taktaev-artyom — APPROVED** [review]
for Maksim_Shagov

**Review by streltsova-yana — APPROVED** [review]

**Pull request merged**